test/10-verify.spec.js

```diff
@@ -130,550 +84,793 @@ before(async () => {
+          const ecdsaSdDeriveSuite = new DataIntegrityProof({
+            cryptosuite: createDiscloseCryptosuite({
+              proofId,
+              selectivePointers: [
+                '/credentialSubject'
+              ]
+            })
+          });
+          // setup ecdsa-sd-2023 suite for verifying selective disclosure VCs
+          const ecdsaSdVerifySuite = new DataIntegrityProof({
+            cryptosuite: createVerifyCryptosuite()
+          });
+
+          const verifiableCredential = await vc.issue({
+            credential: jsonld.clone(mockCredential),
+            suite: ecdsaSdSignSuite,
+            documentLoader
+          });
+          const derivedCredential = await vc.derive({
+            verifiableCredential,
+            suite: ecdsaSdDeriveSuite,
+            documentLoader
+          });
+          derivedCredential.credentialSubject.id = `urn:uuid:${uuid()}`;
+          const result = await vc.verifyCredential({
+            credential: derivedCredential,
+            controller: assertionController,
+            suite: ecdsaSdVerifySuite,
+            documentLoader
+          });
+          result.verified.should.be.a('boolean');
+          result.verified.should.be.false;
+        });
       });
-      result.verified.should.be.a('boolean');
-      result.verified.should.be.false;
     });
-  });
-});
-
-describe('verify API (presentations)', () => {
-  it('verifies a valid signed presentation', async () => {
-    const challenge = uuid();
 
-    const {presentation, suite, documentLoader} =
-      await _generatePresentation({challenge});
+    describe('verify API (presentations)', () => {
+      it('verifies a valid signed presentation', async () => {
+        const challenge = uuid();
+
+        const {presentation, suite, documentLoader} =
+          await _generatePresentation({challenge, mockCredential, version});
+
+        const result = await vc.verify({
+          challenge,
+          suite,
+          documentLoader,
+          presentation
+        });
+
+        if(result.error) {
+          const firstError = [].concat(result.error)[0];
+          throw firstError;
+        }
+        result.verified.should.be.a('boolean');
+        result.verified.should.be.true;
+      });
 
-    const result = await vc.verify({
-      challenge,
-      suite,
-      documentLoader,
-      presentation
+      it('verifies an unsigned presentation', async () => {
+        const {
+          presentation,
+          suite: vcSuite,
+          documentLoader
+        } = await _generatePresentation({
+          unsigned: true,
+          mockCredential,
+          version
+        });
+
+        const result = await vc.verify({
+          documentLoader,
+          presentation,
+          suite: vcSuite,
+          unsignedPresentation: true
+        });
+
+        if(result.error) {
+          const firstError = [].concat(result.error)[0];
+          throw firstError;
+        }
```

Comment:
This is actually not my code or part of this PR, but I do know that the VC library doesn't throw in all cases, verification can return an error or an array of errors. it looks like this code takes that error which could be an array, ensures it is an array and then throws the first error. We probably should be asserting on that error. Maybe an issue can be made about this to assert on these errors rather than throw?

Reply:
@dmitrizagidulin got any insight here? it looks like you just wanted to throw if there is an error so you could check it and figure out what was going wrong with this lib?